### PR TITLE
Add 'Sink.reload' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1597 as a base.~~

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merge of #1597 to `logging-api` branch.  PTAL

File: gcloud/logging/sink.py [hunk 1]
```diff
@@ -111,3 +111,18 @@ def exists(self, client=None):
...
+        https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks/get
+
+        :type client: :class:`gcloud.logging.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current sink.
+        """
+        client = self._require_client(client)
+        data = client.connection.api_request(method='GET', path=self.path)
+        self.filter_ = data['filter']
+        self.destination = data['destination']
```
Comment (dhermes) [inline gcloud/logging/sink.py]: Not worried about checking the other two fields in the payload?

Comment (tseaver) [inline gcloud/logging/sink.py]: - The `name` field is redundant to the URL used to fetch the payload, which derives from `self.name`.   A mismatch would indicate a bug in the back-end.
- We don't expose the `outputVersionFormat` anywhere: it defaults to `VERSION_FORMAT_UNSPECIFIED`, i.e. `V2`)  If we decide to allow the user to set / query it on the `Sink` instance, then we can pull it out here.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**